src/Chapter2.hs

```diff
@@ -842,7 +876,17 @@ list.
 
 🕯 HINT: Use the 'cycle' function
 -}
-rotate = error "rotate: Not implemented!"
+rotate :: Int -> [a] -> [a]
+rotate n ls = if n < 0 then [] else
+  let  
+    infiniteRepetition [] = infiniteRepetition ls
+    infiniteRepetition (x : xs) = x : infiniteRepetition xs
+
+    l = length  ls
+  in take l (drop n (infiniteRepetition ls))
```

Comment:
That's a correct solution! Though, it does a little bit more work when the given number is big and the list is small. You will cycle through the list a lot. Maybe you can think of a way to optimize this :slightly_smiling_face: